src/lem/eval.rs

```diff
@@ -78,11 +79,15 @@ mod tests {
             all_paths.extend(paths);
         }
 
-        lem.assert_all_paths_taken(&all_paths);
+        // TODO: add a case for each tag for this to work
+        // lem.assert_all_paths_taken(&all_frames, store);
```

Comment:
```suggestion
        lem.assert_all_paths_taken(&all_paths);
```
This line is meant to force us into always covering all inputs needed to explore all paths. I would feel more comfortable if we could follow one of the two paths:
* Add match arms only for `Num`, `Nil` and `Str` (since the test cases only cover these tags)
* Expand the test cases to contemplate all paths available
This line was meant not to be skipped as it's going to be an important discipline, especially when the LEM gets bigger.

This is my input, but @porcuquine may have a different take

Reply:
I agree that having some test somewhere that ensure every path has been exercised is high-value. Complementing that, we should make sure that we actually leverage this coverage to check things we should. These would include circuit uniformity, and expectations about slot allocation (for instance, maximum number of slots of each type), etc.